publisher/webhookPublisher.go

```diff
@@ -116,6 +125,9 @@ func (p *WebhookPublisher) makeHttpRequest(r *publishRequest) {
 		})
 	} else {
 		log.Printf("Final retry failed, giving up on %v", url)
+		if r.respChan != nil {
+			r.respChan <- StatusFunctionFailure
+		}
 		// Event dropped
 	}
```

Comment:
it's not super clear but do we guarantee publishing a response into the response channel?